src/module/item/entity.js

```diff
@@ -84,29 +84,33 @@ export class OseItem extends Item {
     };
 
     if (itemData.missile && itemData.melee && !isNPC) {
-      // Dialog
-      new Dialog({
-        title: "Choose Attack Range",
-        content: "",
-        buttons: {
-          melee: {
-            icon: '<i class="fas fa-fist-raised"></i>',
-            label: "Melee",
-            callback: () => {
-              this.actor.targetAttack(rollData, "melee", options);
+      if (options.preferedType === "melee" || options.preferedType === "missile"){
```

Comment:
I wonder if there's a way we could express this without nested if statements?

Reply:
done